### PR TITLE
Add example streaming pipeline server

### PR DESCRIPTION
**Pull request by aajtodd**
This PR adds a very simple pipeline streaming example.

I wanted to put my thoughts out there on how to introduce concepts from tokio-proto without thoroughly overwhelming someone with the kitchen sink. I think we can do that by building small targeted examples with increasing complexity.

This example primarily introduces how to create responses with and without body streams. 

I would expect this crate to build on this example (or another similar simple example if this one is not good enough) to introduce: 
1. A streaming `Transport` = `Stream + Sink + Transport` (`Framed` gives us `Stream + Sink` with a blanket impl of `Transport`)
    * In other words `Framed` is easy mode, you can either wrap a `Framed` instance to implement the Transport trait (for example to make use of the `tick` or `cancel` methods) or implement all three traits.
2. Advanced starting and stopping/cancellation of streams
3. Multiplex streaming examples

I would like to add a client example to go with this using tokio but hit a compiler error using `TcpClient` I have yet to revisit.

In the meantime here is a go example [client](https://gist.github.com/aajtodd/4e201c0cc58f882e242643e1562efeb1) I used to test this server and understand some of the abstractions.

### CODE REVIEW

**Comment by carllerche** [comment]
Thanks for the PR. As I mentioned (way back) in Gitter, I think this example is a bit too intricate to be included here. Maybe it would make more sense as a blog post?
